src/SignalR/server/StackExchangeRedis/src/RedisHubLifetimeManager.cs

```diff
@@ -388,6 +395,78 @@ public void Dispose()
         _ackHandler.Dispose();
     }
 
+    /// <inheritdoc/>
+    public override async Task<T> InvokeConnectionAsync<T>(string connectionId, string methodName, object?[] args, CancellationToken cancellationToken = default)
+    {
+        // send thing
+        if (connectionId == null)
+        {
+            throw new ArgumentNullException(nameof(connectionId));
+        }
+
+        var connection = _connections[connectionId];
+
+        // Needs to be unique across servers, easiest way to do that is prefix with connection ID.
+        var invocationId = $"{connectionId}{Interlocked.Increment(ref _lastInvocationId)}";
+
+        using var _ = CancellationTokenUtils.CreateLinkedToken(cancellationToken,
+            connection?.ConnectionAborted ?? default, out var linkedToken);
+        var task = _clientResultsManager.AddInvocation<T>(connectionId, invocationId, linkedToken);
+
+        try
+        {
+            if (connection == null)
+            {
+                // TODO: Need to handle other server going away while waiting for connection result
+                var m = _protocol.WriteInvocation(methodName, args, invocationId, returnChannel: _channels.ReturnResults(_serverName));
+                var received = await PublishAsync(_channels.Connection(connectionId), m);
+                if (received < 1)
+                {
+                    throw new InvalidOperationException("Connection does not exist.");
```

Comment:
It's super cool we can fail early in this case. Let's include the connectionId in the exception. Also considering this is most likely going to be caused by a client disconnecting, it should probably be an `IOException`. It should be possible to always avoid `InvalidOperationException`s with the right code, but it's not here because of the race.